activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -251,7 +253,7 @@ def create_database(name, options = {})
         if options[:collation]
           execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT COLLATE #{quote_table_name(options[:collation])}"
         else
-          execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT CHARACTER SET #{quote_table_name(options[:charset] || 'utf8')}"
+          execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT CHARACTER SET #{quote_table_name(options[:charset] || 'utf8mb4')}"
```

Comment:
I noticed that [lib/active_record/tasks/mysql_database_tasks.rb](https://github.com/rails/rails/blob/53be0132c3f66dd8510921215d940641bbfd9588/activerecord/lib/active_record/tasks/mysql_database_tasks.rb#L77-L82) is using `config[:encoding]` as default charset, whereas we fall back to utf8mb4 here regardless of configured encoding.

```ruby
def creation_options
  Hash.new.tap do |options|
    options[:charset]     = configuration["encoding"]   if configuration.include? "encoding"
```

It's preexisting behavior, but should we do the same here? e.g. `options[:charset] || @config[:encoding] || 'utf8mb4'`